input/src/main/scala/Cats1.scala

```diff
@@ -0,0 +1,335 @@
+package cats
+
+import cats.ApplicativeError.CatchOnlyPartiallyApplied
+import cats.data.{EitherT, Validated}
+import cats.data.Validated.{Invalid, Valid}
+
+import scala.reflect.ClassTag
+import scala.util.control.NonFatal
+import scala.util.{Failure, Success, Try}
+
+/**
+ * An applicative that also allows you to raise and or handle an error value.
+ *
+ * This type class allows one to abstract over error-handling applicatives.
+ */
+trait ApplicativeError[F[_], E] extends Applicative[F] {
```

Comment:
Indeed this is a lot of code? How hard would it be to remove the pieces that are not relevant?

Reply:
not that hard .. it was just to see how the dichotomic approach works.